core/images/archive/importer.go

```diff
@@ -82,6 +80,12 @@ func ImportIndex(ctx context.Context, store content.Store, reader io.Reader, opt
 			return ocispec.Descriptor{}, err
 		}
 	}
+	s, err := compression.DecompressStream(reader)
```

Comment:
The interface expects the caller to call `DecompressStream`. This would cause `DecompressStream` to be called twice in some cases.

Reply:
Sorry, May I  do not understand （Can you explain it to me clearly? thanks）.  I add this code ,because when I use ctr i import any file ,not only tar , I can be imported successful. I search the code , but just image Import will call ImportIndex (Is there something I missing?) . If the reader is tar file , the DecompressStream will do nothing , I think it will not cause any problem  if it is called twice.